### PR TITLE
Assignment7 Doug Doner

### PR DESCRIPTION
**Pull request by dougdoner**

### CODE REVIEW

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs [hunk 1]
```diff
@@ -8,6 +9,50 @@ public class SortUtilityTests
         [TestMethod]
         public void SortUtility_ShouldSortAscending_UsingAnAnonymousMethod()
         {
+            //Organize
+            int[] intArr = { 3, 4, 5, 2, 4, 2, 1};
+            SortUtility sortUtil = new SortUtility();
+
+            //Act
+            intArr = sortUtil.Sort(intArr,  delegate(int first, int second)
```
Comment (RSXEnthusiast) [inline DelegateFun/Sorter.Tests/SortUtilityTests.cs]:
```suggestion
            intArr = Sort(intArr,  delegate(int first, int second)
```

Save some typing if you just import Sorter.SortUtil at the top instead.

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
```
Comment (RSXEnthusiast) [inline DelegateFun/Sorter.Tests/SortUtilityTests.cs]:
```suggestion

using Sorter.SortUtil;
```

Save some typing if you just import Sorter.SortUtil at the top instead.

File: DelegateFun/Sorter/SortUtility.cs [hunk 3]
```diff
@@ -4,7 +4,30 @@ namespace Sorter
...
+
+            for (int i = array.Length - 1; i >= 0; i--)
+            {
+                for (int j = 1; j <= i; j++)
+                {
+                    if (sortOrder(array[j - 1],array[j]))
+                    {
+                        var temp = array[j - 1];
+                        array[j - 1] = array[j];
+                        array[j] = temp;
```
[the remark this answers is not in the document]
Comment (dougdoner) [reply DelegateFun/Sorter/SortUtility.cs]: Oh shoot I didn't realize that. I will fix that.

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs (hunk 1 above)
Comment (dougdoner) [reply DelegateFun/Sorter.Tests/SortUtilityTests.cs]: Not sure I understand this. I'm instantiating a SortUtility object so I have to call sortUtil.Sort() from the object. Any clarification would be great :)

File: DelegateFun/Sorter/SortUtility.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (dougdoner) [reply DelegateFun/Sorter/SortUtility.cs]: Fixed in 3b87da0

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs [hunk 4]
```diff
@@ -8,6 +9,50 @@ public class SortUtilityTests
...
+            SortUtility sortUtil = new SortUtility();
+
+            //Act
+            intArr = sortUtil.Sort(intArr,  delegate(int first, int second)
+            {
+                return first > second;
+            });
+
+            //Assert
+            Assert.AreEqual(1, intArr[0]);
```
[the remark this answers is not in the document]
Comment (dougdoner) [reply DelegateFun/Sorter.Tests/SortUtilityTests.cs]: Fixed in 3b87da0

File: DelegateFun/Sorter.Tests/SortUtilityTests.cs (hunk 1 above)
Comment (RSXEnthusiast) [reply DelegateFun/Sorter.Tests/SortUtilityTests.cs]: You can just put "using Sorter.SortUtil;" at the top if you want, then you haven't got to type SortUtil every time you call Sort. Honestly doesn't make much of a difference, just a small refactoring suggestion.

File: DelegateFun/Sorter/SortUtility.cs [hunk 5]
```diff
@@ -4,7 +4,29 @@ namespace Sorter
 {
     public class SortUtility
     {
+        public delegate bool SortOrder(int first, int second);
         // Sort method should be implemented here
         // It should accept an int[] and a delegate you define that performs the actual comparison
+        public int[] Sort(int[] array, SortOrder sortOrder)
+        {
+            if(array is null)
```
Comment (breaman) [inline DelegateFun/Sorter/SortUtility.cs]: Where is the test for this?

**Pull request closed without merging**